Add ScrapeConfig project idea (#4)
prometheus-operator/community: mentoring/gsoc/2024/project_ideas.md

---

## Ideas

### Extending ScrapeConfig surface

> - Description: The ScrapeConfig is a recently added CRD of Prometheus-Operator. Differently from other monitoring related CRDs like ServiceMonitor and PodMonitor, ScrapeConfig offers a more generic approach by aiming to be a 1:1 equivalent of the Prometheus scrape configuration. Since ScrapeConfig was created pretty recently, it is still far from being a 1:1 mapping of the Prometheus configuration. For us to be able to promote the ScrapeConfig CRD from alpha to beta/stable, we need to cover more missing fields.

Reviewer: ```suggestion
- Description: The ScrapeConfig is a recently added Custom Resource Definition (CRD) of Prometheus-Operator. Differently from other monitoring related CRDs like ServiceMonitor and PodMonitor, ScrapeConfig offers a more generic approach by aiming to be a 1:1 equivalent of the Prometheus scrape configuration. The current state of ScrapeConfig is far from being feature-complete compared to whaat Prometheus configuration offers. For us to be able to promote the ScrapeConfig CRD from alpha to beta/stable, we need to implement more service discoveries as well as missing configuration options.
```

> - Description: The ScrapeConfig is a recently added Custom Resource Definition (CRD) of Prometheus-Operator. Differently from other monitoring related CRDs like ServiceMonitor and PodMonitor, ScrapeConfig offers a more generic approach by aiming to be a 1:1 equivalent of the Prometheus scrape configuration. The current state of ScrapeConfig is far from being feature-complete compared to whaat Prometheus configuration offers. For us to be able to promote the ScrapeConfig CRD from alpha to beta/stable, we need to implement more service discoveries as well as missing configuration options.

Reviewer 2: ```suggestion
- Description: The ScrapeConfig is a recently added Custom Resource Definition (CRD) of Prometheus-Operator. Differently from other monitoring related CRDs like ServiceMonitor and PodMonitor, ScrapeConfig offers a more generic approach by aiming to be a 1:1 equivalent of the Prometheus scrape configuration. The current state of ScrapeConfig is far from being feature-complete compared to what Prometheus configuration offers. For us to be able to promote the ScrapeConfig CRD from alpha to beta/stable, we need to implement more service discoveries as well as missing configuration options.
```

- Recommended Skills: Go, Prometheus
- Expected project size: small
- Mentors:

>   - Jayapriya Pai (@slashpai, janantha@redhat.com)

Reviewer 2: ```suggestion
  - Jayapriya Pai (@slashpai, slashpai9@gmail.com)
```
